File: Implementacao/backend/src/main/java/com/codelink/dto/PostDTO.java
package com.codelink.dto;

import java.time.LocalDateTime;
import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PostDTO {
    private Long id;
    private String conteudo;
    private String imagem;
    private LocalDateTime dataPostagem;
    private int likes;
    private UsuarioDTO autor;
    private Long comunidadeId;
    private List<ComentarioDTO> comentarios;
    // getters/setters
}

File: Implementacao/frontend/src/app/shared/models/post.dto.ts
import { UsuarioDTO } from "./usuario.dto";
import { ComentarioDTO } from "./comentario.dto";

export interface PostDTO {
  id: number;
  conteudo: string;
  imagem: string | null;
  dataPostagem: string;
  likes: number;
  autor: UsuarioDTO;
  comunidadeId?: number | null;
  comentarios: ComentarioDTO[];
}
